### PR TITLE
feat: add staging workflow for CD

### PR DESCRIPTION
**Pull request by rafaiovlabs**
Staging workflow for CD of marketplace UI

### CODE REVIEW

File: .github/workflows/staging.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: RIF Marketplace UI deploy
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+-**'
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+    branches:
+      - staging
```
Comment (artem-brazhnikov) [inline .github/workflows/staging.yml:9]: ok good, the dev env is approved, I will work on the branches workflow so it's consistent across repos

**Review by artem-brazhnikov — APPROVED** [review]

**Review by jurajpiar — CHANGES_REQUESTED** [review]
Change the commit message to follow the [conventional commit specs](https://www.conventionalcommits.org/en/v1.0.0/) please

**Comment by artem-brazhnikov** [comment]
@jurajpiar @rafaiovlabs we need to find out why the pipeline is failing, I restarted the CI. There seemed to be an out of memory error.

**Comment by artem-brazhnikov** [comment]
@rafaiovlabs can you please check the amount of memory allocated for the CI? I think it might not enough
https://discuss.circleci.com/t/unit-tests-enomem-not-enough-memory-read/35695/3

File: .github/workflows/staging.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: RIF Marketplace UI deploy
+
+on:
+  push:
+    tags:
```
Comment (artem-brazhnikov) [inline .github/workflows/staging.yml:5]: @rafaiovlabs I was thinking through this and I am not sure we need to deploy tags to the staging. Image when we merged some piece of logic to staging and QA are working on it. And then we make a release from master and suddenly the QA are presented with a different version.

**Review by artem-brazhnikov — CHANGES_REQUESTED** [review]
Also, in the previous PR there were jobs for the testnet and mainnet. Why they disappeared in this one?

File: .github/workflows/staging.yml (hunk 2 above)
Comment (rafaiovlabs) [reply .github/workflows/staging.yml:5]: @artem-brazhnikov  Fixed the mem issue of CI with this PR, about the tags..i thing those give us better control (in dev environment they are not needed at all) , but for staging could be of benefit and also we can have tags like v2.3.0-qafix or something like that. About not having CD for other envs thats correct i only have staging precisely to validate the deployment, workflows and so on, after validation CD for other envs is trivial to add.

Comment (artem-brazhnikov) [reply .github/workflows/staging.yml:5]: ok, good

Comment (artem-brazhnikov) [reply .github/workflows/staging.yml:5]: ok, lets try and see how it goes

**Review by artem-brazhnikov — APPROVED** [review]

**Pull request merged**